src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -249,7 +249,7 @@ class Ledger(vm: VM) extends Logger {
   private[ledger] def updateSenderAccountBeforeExecution(stx: SignedTransaction, worldStateProxy: InMemoryWorldStateProxy): InMemoryWorldStateProxy = {
     val senderAddress = stx.senderAddress
     val account = worldStateProxy.getGuaranteedAccount(senderAddress)
-    worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontCost(stx.tx)).increaseNonce)
+    worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontGas(stx.tx)).increaseNonce)
```

Comment:
Yes, it is currently used in the `validateAccountHasEnoughGasToPayUpfrontCost` function to validate that the account has enough ether to pay for the tx execution. However I agree that it would be better to have the transfer not done in `ProgramContext` it also confused me that it was done that way.